lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -373,10 +377,11 @@ void SPIRVRegularizeLLVMBase::adaptStructTypes(StructType *ST) {
   // simply not true.
   if (MangledName == "__spirv_JointMatrixINTEL") {
```

Comment:
Guess, it should be looking like this then, SPIRVWriter part is supposed to parse the type name:

```suggestion
  if (MangledName == "__spirv_JointMatrixINTEL" && !PtrTy->isOpaque()) {
```